javascript/app-discover/src/utils/CausalEdgeReport.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+
+import { hasAnyConstraint } from '../components/lists/EdgeList.utils.js'
+import { CausalDiscoveryAlgorithm } from '../domain/CausalDiscovery/CausalDiscoveryAlgorithm.js'
+import type { CausalDiscoveryConstraints } from '../domain/CausalDiscovery/CausalDiscoveryConstraints.js'
+import type { CausalEdgesReport } from '../domain/CausalEdgesReport.js'
+import { VariableNature } from '../domain/VariableNature.js'
+import type { ATEDetailsByName } from './../domain/CausalDiscovery/CausalDiscoveryResult.js'
+import type {
+	Relationship,
+	RelationshipWithWeight,
+} from './../domain/Relationship.js'
+import { correlationForVariables } from './Correlation.js'
+
+export function row2report(
+	relationship: Relationship,
+	selectedCausalDiscoveryAlgorithm: CausalDiscoveryAlgorithm,
+	constraints: CausalDiscoveryConstraints,
+	correlations: RelationshipWithWeight[],
+	ATEDetailsByName?: ATEDetailsByName,
+): CausalEdgesReport {
+	const hasConstraints = hasAnyConstraint(constraints, relationship)
+	const isPC = selectedCausalDiscoveryAlgorithm === CausalDiscoveryAlgorithm.PC
+	const correlation = correlationForVariables(
+		correlations,
+		relationship.source,
+		relationship.target,
+	)
+	const ate =
+		ATEDetailsByName && ATEDetailsByName[relationship?.source.columnName]
+	const isNatureCategorical = isCategorical(ate?.nature)
+
+	return {
```

Comment:
Since there are some differeneces in the report by algorithm, I've been thinking about creating a reportGenerator that would build what we need based on the type instead of a lot of conditionals...

Reply:
Maybe consider a ReportGenerator class that handles most of it via base methods, and each algo can optionally have an extension, then use a factory to return the correct class instance

Reply:
good idea, thanks!